src/protocol/contract.rs

```diff
@@ -836,4 +836,189 @@ mod test {
         )
         .is_ok());
     }
+
+    #[test]
+    fn test_check_multisig_has_pubkey() {
```

Comment:
This is I am noticing right now. Usually the name does not need to contain the word `test`. Its implied. All the names in test names in this file could be updated to remove the `test` prefix.

Same thing for other files, but can be done as later PR.